NPE on  map

New crash report with 2019.02.17-RC. 
Not sure if this can be related to sdk-raise or not.

Device: Android 9.0

Crash report:
```
java.lang.NullPointerException: 
 
  at io.reactivex.internal.functions.ObjectHelper.requireNonNull (ObjectHelper.java:39)
 
  at io.reactivex.Observable.combineLatest (Observable.java:433)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:111)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:77)
 
  at cgeo.geocaching.maps.mapsforge.v6.NewMap.initializeLayers (NewMap.java:784)
 
  at cgeo.geocaching.maps.mapsforge.v6.NewMap.onStart (NewMap.java:725)
 
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1392)
 
  at android.app.Activity.performStart (Activity.java:7157)
 
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:2952)
 
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:180)
 
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:165)
 
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:142)
 
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1823)
 
  at android.os.Handler.dispatchMessage (Handler.java:107)
 
  at android.os.Looper.loop (Looper.java:198)
 
  at android.app.ActivityThread.main (ActivityThread.java:6729)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
```

There is a similar NPE reported also for 2019.02.17-RC but seems related to Google Maps rather then new map:

```
java.lang.RuntimeException: 
 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3844)
 
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3885)
 
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3051)
 
  at android.app.ActivityThread.-wrap14 (ActivityThread.java)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1642)
 
  at android.os.Handler.dispatchMessage (Handler.java:102)
 
  at android.os.Looper.loop (Looper.java:154)
 
  at android.app.ActivityThread.main (ActivityThread.java:6776)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1518)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1408)
Caused by: java.lang.NullPointerException: 
 
  at io.reactivex.internal.functions.ObjectHelper.requireNonNull (ObjectHelper.java:39)
 
  at io.reactivex.Observable.combineLatest (Observable.java:433)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:111)
 
  at cgeo.geocaching.sensors.GeoDirHandler.start (GeoDirHandler.java:77)
 
  at cgeo.geocaching.maps.CGeoMap.onResume (CGeoMap.java:528)
 
  at cgeo.geocaching.maps.google.v1.GoogleMapActivity.onResume (GoogleMapActivity.java:68)
 
  at android.app.Instrumentation.callActivityOnResume (Instrumentation.java:1277)
 
  at android.app.Activity.performResume (Activity.java:7115)
 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:3821)
```

--- a/main/src/cgeo/geocaching/maps/CGeoMap.java
+++ b/main/src/cgeo/geocaching/maps/CGeoMap.java
@@ -31,6 +31,9 @@ import cgeo.geocaching.maps.routing.RoutingMode;
 import cgeo.geocaching.models.Geocache;
 import cgeo.geocaching.models.Waypoint;
 import cgeo.geocaching.network.AndroidBeam;
+import cgeo.geocaching.permission.PermissionGrantedCallback;
+import cgeo.geocaching.permission.PermissionHandler;
+import cgeo.geocaching.permission.PermissionRequestContext;
 import cgeo.geocaching.sensors.GeoData;
 import cgeo.geocaching.sensors.GeoDirHandler;
 import cgeo.geocaching.sensors.Sensors;
@@ -525,7 +528,20 @@ public class CGeoMap extends AbstractMap implements ViewFactory {
     @Override
     public void onResume() {
         super.onResume();
-        resumeDisposables.addAll(geoDirUpdate.start(GeoDirHandler.UPDATE_GEODIR), startTimer());
+
+        // resume location access
+        PermissionHandler.executeIfLocationPermissionGranted(this.activity, new PermissionGrantedCallback(PermissionRequestContext.CacheDetailActivity) {
+
+            @Override
+            public void execute() {
+                Log.d("CGeoMap.OnResumePermissionGrantedCallback.execute");
+                final Sensors sensors = Sensors.getInstance();
+                sensors.setupGeoDataObservables(Settings.useGooglePlayServices(), Settings.useLowPowerMode());
+                sensors.setupDirectionObservable();
+
+                resumeDisposables.addAll(geoDirUpdate.start(GeoDirHandler.UPDATE_GEODIR), startTimer());
+            }
+        });
 
         final List<String> toRefresh;
         synchronized (dirtyCaches) {

--- a/main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
+++ b/main/src/cgeo/geocaching/maps/mapsforge/v6/NewMap.java
@@ -38,6 +38,9 @@ import cgeo.geocaching.maps.mapsforge.v6.layers.TapHandlerLayer;
 import cgeo.geocaching.maps.routing.Routing;
 import cgeo.geocaching.maps.routing.RoutingMode;
 import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.permission.PermissionGrantedCallback;
+import cgeo.geocaching.permission.PermissionHandler;
+import cgeo.geocaching.permission.PermissionRequestContext;
 import cgeo.geocaching.sensors.GeoData;
 import cgeo.geocaching.sensors.GeoDirHandler;
 import cgeo.geocaching.sensors.Sensors;
@@ -781,7 +784,19 @@ public class NewMap extends AbstractActionBarActivity implements XmlRenderThemeM
             targetView.setTarget(target.getGeocode(), target.getName());
         }
 
-        this.resumeDisposables.add(this.geoDirUpdate.start(GeoDirHandler.UPDATE_GEODIR));
+        // resume location access
+        PermissionHandler.executeIfLocationPermissionGranted(this, new PermissionGrantedCallback(PermissionRequestContext.CacheDetailActivity) {
+
+            @Override
+            public void execute() {
+                Log.d("NewMap.initializeLayersPermissionGrantedCallback.execute");
+                final Sensors sensors = Sensors.getInstance();
+                sensors.setupGeoDataObservables(Settings.useGooglePlayServices(), Settings.useLowPowerMode());
+                sensors.setupDirectionObservable();
+
+                resumeDisposables.add(geoDirUpdate.start(GeoDirHandler.UPDATE_GEODIR));
+            }
+        });
     }
 
     @Override

